cosine/tests.py

```diff
@@ -44,3 +46,106 @@ def test_call_detail_loads(self):
     def test_call_detail_returns_404_for_nonexistent_website(self):
         response = self.client.get('/{}'.format(10000))
         self.assertEqual(response.status_code, 404)
+
+    def test_dashboard_redirect_if_not_logged_in(self):
+        response = self.client.get('/', follow=True)
+        self.assertEqual(response.status_code, 200)
+        self.assertTemplateUsed(response, 'registration/login.html')
+
+    def test_dashboard_renders_if_logged_in(self):
+        self.client.login(username='user1', password='12345')
+        response = self.client.get('/')
+        self.assertEqual(response.status_code, 200)
+        self.assertTemplateUsed(response, 'cosine/dashboard.html')
+
+    def test_user_login_view_if_not_succes(self):
+        self.client.login(username='user1', password='12345')
+        self.client.get('/logout')
+        response = self.client.post('/login/', {'username':'user1', 'password':'112345'})
+        self.assertContains(response, 'Incorrect username or password')
+
+    def test_user_login_view_if_succes(self):
+        self.client.login(username='user1', password='12345')
+        self.client.get('/logout')
+        response = self.client.post('/login/', {'username':'user1', 'password':'12345'}, follow=True)
+        self.assertTemplateUsed(response, 'cosine/dashboard.html')
+
+    def test_register_view_if_succes(self):
+        response = self.client.get('/register', follow=True)
+        self.assertTemplateUsed(response, 'cosine/register.html')
+        response = self.client.post('/register/', {'username':'user2',
+                                                'first_name':'Aaa',
+                                                'last_name':'Bbb',
+                                                'email':'aaa@bbb.com',
+                                                'password2':'12345',
+                                                'password':'12345'}, follow=True)
+        self.assertTemplateUsed(response, 'cosine/register_done.html')
+
+    def test_register_view_if_not_succes(self):
```

Comment:
Podziel prosze na 3 oddzielne testy, jak robisz assert to najlepiej juz wiecej niz nie testować. Jak ktorys assert sie nie uda, to reszta testów sie w tym samym tescie sie nie wykona.